pkg/sysregistriesv2/shortnames.go

```diff
@@ -0,0 +1,293 @@
+		return err
+	}
+
+	// If the name does not exist, throw an error.
+	if _, exists := shortNameConf.Aliases[name]; !exists {
+		return errors.Errorf("short-name alias %q not found in %q: please check registries.conf files", name, confPath)
+	}
+
+	delete(shortNameConf.Aliases, name)
+
+	f, err := os.OpenFile(confPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	encoder := toml.NewEncoder(f)
+	return encoder.Encode(shortNameConf)
+
+}
+
+// parseAliase parses the specified alias into a reference.Named.  The alias is
+// expected to not be tagged or carry a digest and *must* include a
+// domain/registry.
+func parseAlias(alias string) (reference.Named, error) {
+	repo, err := reference.Parse(alias)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot parse alias: %q", alias)
+	}
+
+	if _, ok := repo.(reference.Digested); ok {
+		return nil, errors.Errorf("invalid alias %q: must not contain digest", alias)
+	}
+
+	if _, ok := repo.(reference.Tagged); ok {
+		return nil, errors.Errorf("invalid alias %q: must not contain tag", alias)
+	}
+
+	named, ok := repo.(reference.Named)
+	if !ok {
+		return nil, errors.Errorf("invalid alias %q: must contain registry and repository", alias)
+	}
+
+	registry := reference.Domain(named)
+	if !(strings.ContainsAny(registry, ".:") || registry == "localhost") {
+		return nil, errors.Errorf("invalid alias %q: must contain registry and repository", alias)
+	}
+	return named, nil
+}
+
+// validateShortName parses the specified `name` of an alias (i.e., the left-hand
+// side) and checks if it's a short name and does not include a tag or digest.
+func validateShortName(name string) error {
+	repo, err := reference.Parse(name)
+	if err != nil {
+		return errors.Wrapf(err, "cannot parse short name: %q", name)
+	}
+
+	if _, ok := repo.(reference.Digested); ok {
+		return errors.Errorf("invalid short name %q: must not contain digest", name)
+	}
+
+	if _, ok := repo.(reference.Tagged); ok {
+		return errors.Errorf("invalid short name %q: must not contain tag", name)
+	}
+
+	named, ok := repo.(reference.Named)
+	if !ok {
+		return nil
+	}
+
+	registry := reference.Domain(named)
+	if strings.ContainsAny(registry, ".:") || registry == "localhost" {
+		return errors.Errorf("invalid short name %q: must not contain registry", name)
+	}
+	return nil
+}
+
+// parseAndValidate parses and validates all entries in conf.Aliases and stores
+// the results in conf.namedAliases.
+func (conf *shortNameAliasConf) parseAndValidate() error {
+	if conf.namedAliases == nil {
+		conf.namedAliases = make(map[string]reference.Named)
+	}
+	errs := []error{}
+	for name, alias := range conf.Aliases {
+		if err := validateShortName(name); err != nil {
+			errs = append(errs, err)
+		}
+
+		named, err := parseAlias(alias)
+		if err != nil {
+			// We want to report *all* malformed entries to avoid a
+			// whack-a-mole for the user.
+			errs = append(errs, err)
+		}
+		if len(errs) == 0 {
```

Comment:
Non-blocking: I’d just use `else` here, the efficiency in an error case is not worth the mental overhead of figuring out that `len(errs)` implies `err == nil`, to me.